File: sdk/python/compliagl/clients/intents.py
from __future__ import annotations
from .base import CrudClient


class IntentClient(CrudClient):
    path = "intents"

    def transition(self, id: str, status: str, *, idempotency_key: str | None = None):
        return self._post(f"{self.path}/{id}/transition", {"status": status}, idempotency_key=idempotency_key)

File: sdk/typescript/src/clients/intents.ts
import { CrudClient } from './base';
import type { Intent, CreateIntent } from '../models';
import type { RequestOptions } from '../config';
export class IntentClient extends CrudClient<Intent, CreateIntent> {
  constructor(http: import('../http').HttpClient) { super(http, '/intents'); }
  transition(id: string, input: { status: string; metadata?: Record<string, unknown> }, options?: RequestOptions): Promise<Intent> { return this.http.post<Intent>(`/intents/${encodeURIComponent(id)}/transition`, input, options); }
}
